### PR TITLE
[3006.x] Fix lgpo.get functions not returning all boolean values

### LINKED ISSUE #63296: [BUG] lgpo.set fails when policy is already in desired state
**Description**
If the group policy already configured as declared in state file, Salt state.apply execution exits with failure.

Note: This behaviour is not limited to the policy given in example below - there are multiple policies(from ADMX template referenced below) with described bug. 

**Setup**

State file hardening-microsoft-test.sls:
```yaml
'File Type Blocking Excel':
  lgpo.set:
    - user_policy:
         "Microsoft Excel 2016\\Excel Options\\Security\\Trust Center\\File Block Settings\\Dif and Sylk files":
           "L_DifAndSylkFilesDropID": "Open/Save blocked, use open policy"
```

**Steps to Reproduce the behavior**
1. Get Administrative Template files (ADMX/ADML) for Microsoft 365 Apps for enterprise/Office LTSC 2021/Office 2019/Office 2016 and the Office Customization Tool for Office 2016: https://www.microsoft.com/en-us/download/details.aspx?id=49030
2. Copy templates to Windows machine - upload ADMX files to  'C:\Windows\PolicyDefinitions\', restart PC. If applied successfully, you should see Office the policies in Local Group Policy Editor: User Configuration\Microsoft Excel 2016\Excel Options\Security\Trust Center\File Block Settings\Dif and Sylk files
3. Apply the state more than once:
```yaml
$sudo salt 'test-S*' state.apply security.hardening-microsoft-test
test-Surface-cracked:
----------
          ID: File Type Blocking Excel
    Function: lgpo.set
      Result: True
     Comment: The following policies changed:
              Microsoft Excel 2016\Excel Options\Security\Trust Center\File Block Settings\Dif and Sylk files
     Started: 20:08:01.484804
    Duration: 24129.009 ms
     Changes:   
              ----------
              new:
                  ----------
                  User Configuration:
                      ----------
                      Microsoft Excel 2016\Excel Options\Security\Trust Center\File Block Settings\Dif and Sylk files:
                          ----------
                          File block setting:
                              Open/Save blocked, use open policy
              old:
                  ----------
                  User Configuration:
                      ----------
                      Microsoft Excel 2016\Excel Options\Security\Trust Center\File Block Settings\Dif and Sylk files:
                          Not Configured

Summary for test-Surface-cracked
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  24.129 s
```
run second time:
```yaml
$sudo salt 'test-S*' state.apply security.hardening-microsoft-test
test-Surface-cracked:
----------
          ID: File Type Blocking Excel
    Function: lgpo.set
      Result: False
     Comment: Failed to set the following policies:
              Microsoft Excel 2016\Excel Options\Security\Trust Center\File Block Settings\Dif and Sylk files
     Started: 20:16:33.973043
    Duration: 24291.846 ms
     Changes:   

Summary for test-Surface-cracked
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  24.292 s
```

**Expected behavior**
Salt should exit with "Result: True" and notify there are no changes applied in current state of configured GPO.

**Versions Report**
MASTER:
$ salt -V
Salt Version:
          Salt: 3005.1
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.0.7
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: Not Installed
        Python: 3.8.10 (default, Nov 14 2022, 12:59:47)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 20.0.0
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2
 
System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.15.0-1023-aws
        system: Linux
       version: Ubuntu 20.04 focal

MINION:
    Salt Minion 3005.1 (Tiamat):
        3005.1
 
   os_name:
        Microsoft Windows 10 Pro
    os_type:
        Work Station
    os_version:
        10.0.19045

</details>

### LINKED ISSUE #63473: [BUG] lgpo.set fails to set bool elements when mixed in with enums in 3005.1
**Description**

When using `lgpo.set` to set a mix of `enum` and `bool` elements only the enum elements are set. This was working properly in Salt 3004.1. For example:

```yaml
configure_automatic_updates:
  lgpo.set:
    - computer_policy:
        'Configure Automatic Updates':
          'Configure automatic updating': 4 - Auto download and schedule the install
          'Install during automatic maintenance': False
          'Scheduled install day': 0 - Every day
          'Scheduled install time': 03:00
          'Install updates for other Microsoft products': False
          'Every week': False
          'First week of the month': False
          'Second week of the month': False
          'Third week of the month': False
          'Fourth week of the month': False
```

Would result in:

```yaml
          ID: configure_automatic_updates                                                                                
    Function: lgpo.set                                                                                                                                                                        
      Result: True                                                                                                                                                                            
     Comment: The following policies changed:                                                                                                                                                 
              Configure Automatic Updates                                                                                                                                                     
     Started: 08:10:59.720845                                                                                                                                                                 
    Duration: 1141.429 ms                                                                                                                                                                     
     Changes:                                                                                                                                                                                 
              ----------                                                                                                                                                                      
              new:                                                                                                                                                                            
                  ----------                                                                                                                                                                  
                  Computer Configuration:                                                                                                                                                     
                      ----------                                                                                                                                                              
                      Configure Automatic Updates:                                                                                                                                            
                          ----------                                                                                                                                                          
                          Configure automatic updating:                                                                                                                                       
                              4 - Auto download and schedule the install                                                                                                                      
                          Scheduled install day:                                                                                                                                              
                              0 - Every day                                                                                                                                                   
                          Scheduled install time:                                                                                                                                             
                              03:00                                                                                                                                                           
              old:                                                                                                                                                                            
                  ----------                                                                                                                                                                  
                  Computer Configuration:                                                                                                                                                     
                      ----------                                                                                                                                                              
                      Configure Automatic Updates:                                                                                                                                            
                          Not Configured                                                                                                                                                      
```

**Setup**
- [ ] Virtualbox Windows Server 2016
- [ ] Salt Version 3005.1

**Steps to Reproduce the behavior**

SLS file `configure_automatic_updates.sls`:
```yaml
configure_automatic_updates:
  lgpo.set:
    - computer_policy:
        'Configure Automatic Updates':
          'Configure automatic updating': 4 - Auto download and schedule the install
          'Install during automatic maintenance': False
          'Scheduled install day': 0 - Every day
          'Scheduled install time': 03:00
          'Install updates for other Microsoft products': False
          'Every week': False
          'First week of the month': False
          'Second week of the month': False
          'Third week of the month': False
          'Fourth week of the month': False
```

`salt 'window-minions.*' state.apply configure_automatic_updates`

**Expected behavior**

`lgpo.set` should set both `enum` and `bool` elements on the policy.

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:                                                                                                                   
          Salt: 3005.1                                                                                                          
                                                                                                                                
Dependency Versions:                                                                                                            
          cffi: 1.14.6                                                                                                          
      cherrypy: 18.6.1                                                                                                          
      dateutil: 2.8.1                                                                                                           
     docker-py: Not Installed                                                                                                   
         gitdb: 4.0.7                                                                                                           
     gitpython: Not Installed                                                                                                   
        Jinja2: 3.1.0                                                                                                           
       libgit2: Not Installed                                                                                                   
      M2Crypto: Not Installed                                                                                                   
          Mako: 1.1.4                                                                                                           
       msgpack: 1.0.2                                                                                                           
  msgpack-pure: Not Installed                                                                                                   
  mysql-python: Not Installed                                                                                                   
     pycparser: 2.21                                                                                                            
      pycrypto: Not Installed                                                                                                   
  pycryptodome: 3.10.1                                                                                                          
        pygit2: Not Installed                                                                                                   
        Python: 3.8.15 (tags/v3.8.15:44adf8a, Nov  8 2022, 17:20:07) [MSC v.1929 64 bit (AMD64)]                                
  python-gnupg: 0.4.8                                                                                                           
        PyYAML: 5.4.1                                                                                                           
         PyZMQ: 22.0.3                                                                                                          
         smmap: 4.0.0                                                                                                           
       timelib: 0.2.4                                                                                                           
       Tornado: 4.5.3                                                                                                           
           ZMQ: 4.3.4                                                                                                           
                                                                                                                                
System Versions:                                                                                                                
          dist:                                                                                                                 
        locale: cp1252                                                                                                          
       machine: AMD64                                                                                                           
       release: 2016Server                                                                                                      
        system: Windows                                                                                                         
       version: 2016Server 10.0.14393 SP0 Multiprocessor Free                                                                   
                                                                                                                                
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes an issue with the LGPO execution module where not all boolean values were being returned. This was affecting the ``lgpo.set`` state that was expecting to see the boolean settings present, but were instead missing.

The problem had to do with the eTree Element object. Sometimes it is an Element with a length of 0 that would not be reported. Other times it is an Element with a length greater than 0, which would be reported.

### What issues does this PR fix or reference?
Fixes:
- https://github.com/saltstack/salt/issues/63296
- https://github.com/saltstack/salt/issues/63473

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by Ch3LL — DISMISSED** [review]

File: tests/pytests/unit/modules/win_lgpo/test_get_policy.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    policy_name = "Configure Corporate Windows Error Reporting"
+    policy_settings = {
+        "Connect using SSL": False,
+        "Corporate server name": "fakeserver.com",
+        "Only upload on free networks": False,
+        "Server port": 1273,
+    }
+    win_lgpo.set_computer_policy(name=policy_name, setting=copy.copy(policy_settings))
+    yield policy_name, policy_settings
+    win_lgpo.set_computer_policy(name=policy_name, setting="Not Configured")
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/win_lgpo/test_get_policy.py:72]:
```suggestion
    win_lgpo.set_computer_policy(name=policy_name, setting=copy.copy(policy_settings))
    try:
        yield policy_name, policy_settings
    finally:
        win_lgpo.set_computer_policy(name=policy_name, setting="Not Configured")
```
That last function should run even on errors, right?

File: tests/pytests/unit/modules/win_lgpo/test_get_policy.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            "__opts__": {
+                "cachedir": str(cachedir),
+            },
+        },
+        win_file: {
+            "__utils__": {
+                "dacl.set_perms": win_dacl.set_perms,
+            },
+        },
+    }
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/win_lgpo/test_get_policy.py:42]:
```suggestion
@pytest.fixture
def configure_loader_modules(minion_opts):
    return {
        win_lgpo: {
            "__opts__": minion_opts,
            "__salt__": {
                "file.file_exists": win_file.file_exists,
                "file.makedirs": win_file.makedirs_,
            },
        },
        win_file: {
            "__utils__": {
                "dacl.set_perms": win_dacl.set_perms,
            },
        },
    }
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/win_lgpo/test_get_policy.py (hunk 2 above)
Comment (s0undt3ch) [reply tests/pytests/unit/modules/win_lgpo/test_get_policy.py]: Yes, it does. There's also a `master_opts` fixture, in case you ever need one.

**Review by s0undt3ch — APPROVED** [review]

**Comment by bcowman** [comment]
This PR has fixed some GPO Policies, but then I tried these 2 GPO polices using 3007.1.  They work the first time but fail subsequently.

"Interactive logon: Number of previous logons to cache (in case domain controller is not available)"
"Network access: Named Pipes that can be accessed anonymously"

**Pull request merged**